ros2_batch_job/__main__.py

```diff
@@ -356,6 +371,51 @@ def build_and_test(args, job):
     # return 0 if ret_test == 0 and ret_testr == 0 else 1
     return 0
 
+def build_with_sanitizer(cmd, args, job):
+    print('# BEGIN SUBSECTION: build with sanitizer')
+    sanitizer_type = args.enable_sanitizer_type
+    sanitizer_cmake_args = [
+    '-DCMAKE_CXX_FLAGS="-fsanitize=' + sanitizer_type + ' -g"',
+    '-DCMAKE_C_FLAGS="-fsanitize=' + sanitizer_type + ' -g"',
+    '-DCMAKE_EXE_LINKER_FLAGS="-fsanitize=' + sanitizer_type + '"',
+    '-DCMAKE_MODULE_LINKER_FLAGS="-fsanitize=' + sanitizer_type + '"',
+    '-DCMAKE_SHARED_LINKER_FLAGS="-fsanitize=' + sanitizer_type + '"'
+    ]
+    index = cmd.index('--cmake-args')
+    cmd[index + 1:index + 1] = sanitizer_cmake_args
```

Comment:
Not sure I understand what's going there? Is cmd a string? Are we assigning an array to a string and relying on some kind of implicit conversion to string there? Maybe extracting a function and adding a few line of docs would be good to help the reader?

Reply:
cmd here is actually a list of strings which are then concatenated and run as a single command. I have extracted this logic into a separate function and added doc